pyccel/ast/class_defs.py

```diff
@@ -199,35 +199,29 @@
 
 #=======================================================================================
 
-StackArrayClass = ClassDef('stack_array', class_type = CStackArray())
+StackArrayClass = ClassDef('stack_array')
 
 #=======================================================================================
 
 literal_classes = {
-        NativeBool()    : BooleanClass,
-        NativeInteger() : IntegerClass,
-        NativeFloat()   : FloatClass,
-        NativeComplex() : ComplexClass,
-        NativeString()  : StringClass
+        PythonNativeBool()    : BooleanClass,
+        PythonNativeInt()     : IntegerClass,
+        PythonNativeFloat()   : FloatClass,
+        PythonNativeComplex() : ComplexClass,
+        StringType()          : StringClass
 }
 
 #=======================================================================================
 
-def get_cls_base(dtype, precision, container_type):
+def get_cls_base(class_type):
     """
     Determine the base class of an object.
 
-    From the dtype and rank, determine the base class of an object.
+    From the type, determine the base class of an object.
 
     Parameters
     ----------
-    dtype : DataType
-        The data type of the object.
-
-    precision : int
-        The precision of the object.
-
-    container_type : DataType
+    class_type : DataType
         The Python type of the object. If this is different to the dtype then
         the object is a container.
```

Comment:
Maybe this doscstring needs updating? There is no `dtype` any more here.

Reply:
Fixed b01b7774c